src/LayoutManager/LayoutManager.ts

```diff
@@ -0,0 +1,299 @@
+import type { TModificationEvents } from '../EventTypeDefs';
+import { Point } from '../Point';
+import { CENTER, iMatrix } from '../constants';
+import type { Group } from '../shapes/Group';
+import type { FabricObject } from '../shapes/Object/FabricObject';
+import { invertTransform } from '../util/misc/matrix';
+import { resolveOrigin } from '../util/misc/resolveOrigin';
+import { FitContentLayout } from './LayoutStrategies/FitContentLayout';
+import type { LayoutStrategy } from './LayoutStrategies/LayoutStrategy';
+import {
+  LAYOUT_TYPE_INITIALIZATION,
+  LAYOUT_TYPE_ADDED,
+  LAYOUT_TYPE_REMOVED,
+  LAYOUT_TYPE_IMPERATIVE,
+  LAYOUT_TYPE_OBJECT_MODIFIED,
+  LAYOUT_TYPE_OBJECT_MODIFYING,
+} from './constants';
+import type { LayoutContext, LayoutResult, StrictLayoutContext } from './types';
+
+export class LayoutManager {
+  private _prevLayoutStrategy?: LayoutStrategy;
+  private _subscriptions: Map<FabricObject, VoidFunction[]>;
```

Comment:
this should be a WeakMap or we may create memory leaks of some sort.
If the object is used as a key i have no idea if that counts as a reference for the GC, but a WeakMap is probably safer

Reply:
I am not sure this is wise
How will we be able to dispose/unsubscribe?

Reply:
and fixed strokeUniform in group layout
The solution can be used all across fabric to fix stroke uniform issue and remove the wrong `_getTransformedDimensions`

Reply:
a weakMap is still a map, it just does not hold the object in memory if the object gets sent away.
Is not a valid reference to stop garbage collector